### ISSUE TITLE
Mirroring Private GitHub repositories to GitLab

### ISSUE THREAD

**Issue by SethuSenthil** [cooperspencer/gickup#231]
Hey! I am trying to backup/mirror my private GitHub repositories to Gitlab but I'm unable to do so. I get the following error:

``POST https://gitlab.com/api/v4/projects: 422 {message: https://:@github.com/USER-NAME/REPO-NAME.git endpoint error: 401 Unauthorized} stage=gitlab url=https://gitlab.com``

Here is my config yaml:

```YAML

source:
  github:
    - token: TOKEN_STRING_HIDDEN
      ssh: false
      wiki: true
      issues: true
      starred: false
destination:
  gitlab:
    - token: TOKEN_STRING_HIDDEN
```

**Comment by cooperspencer** [comment]
Hi,
I tried that with my account and got the same error message for my private repos on Github. Are the repositories, which cause the issue, private as well?

**Comment by SethuSenthil** [comment]
Yes, only private Github repos have this issue, public Github repos work as intended for Gitlab

**Comment by cooperspencer** [comment]
Ah, you even wrote that it happens for the private repos. Sorry, it is still early on my side :sweat_smile: 

I'll look into it and try to fix it.

**Comment by cooperspencer** [comment]
I found the bug, a two year old bug.
I used the password instead of the token, that should be fixed now.

https://github.com/cooperspencer/gickup/pull/232/files

**Comment by SethuSenthil** [comment]
Np man! Thanks so much for all the work you put in to maintain this free tool!

**Comment by SethuSenthil** [comment]
Oh, I wasn't aware of that. I don't have an enterprise account either :(

**Comment by cooperspencer** [comment]
No problem, I will try to implement the Gitlab mirror feature and get back to you when it is done. It could take me a bit though.

**Comment by SethuSenthil** [comment]
Take your time! Thanks for everything so far!

**Comment by cooperspencer** [comment]
I implemented it now. The changes are in the main branch. Can you test that somehow?
You'd just need to change the following on the gitlab destination.
```yaml
source:
  github:
    - token: TOKEN_STRING_HIDDEN
      ssh: false
      wiki: true
      issues: true
      starred: false
destination:
  gitlab:
    - token: TOKEN_STRING_HIDDEN
      mirror:
         enabled: true
```

**Comment by SethuSenthil** [comment]
Just tested it out on MAIN with the modified config, seems to be working as expected! Thanks a lot!

**Comment by cooperspencer** [comment]
Perfect. A new release should be out within the next few days.

**Comment by SethuSenthil** [comment]
Resolved on #233 and released on [v0.10.30](https://github.com/cooperspencer/gickup/releases/tag/v0.10.30)
